client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,472 @@
+            ) : (
+              <Tab disabled id={tab.id} key={tab.id}>
+                {tab.name}
+              </Tab>
+            )
+          )}
+          <Tabs.Expander />
+          <Button icon="add" minimal disabled>
+            Add Sheet
+          </Button>
+        </Tabs>
+      ) : (
+        // When the above tab bar isn't rendered, still render a hidden input to house the sheet
+        // name. Rendering no input at all interferes with react-hook-form's state management,
+        // specifically its isDirty tracking
+        <input name="name" ref={register()} style={{ display: 'none' }} />
+      )}
+
+      <div
+        // Since we have to render two tab bars, we aren't able to use the tab bar's built-in
+        // `panel` prop and have to manually apply attributes for tab panel accessibility
+        aria-labelledby={`${Classes.TAB_PANEL}_${batch.name}_${selectedTabId}`}
+        role="tabpanel"
+      >
+        <BatchResultTallySheetTable>
+          <thead>
+            <tr>
+              <th>Choice</th>
+              <th>Votes</th>
+            </tr>
+          </thead>
+          <tbody>
+            {contest.choices.map(choice => (
+              <tr key={choice.id}>
+                <td>{choice.name}</td>
+                <td>
+                  {isTotalsSheet ? (
+                    <span>
+                      {sum(sheets.map(sheet => sheet.results[choice.id]))}
+                    </span>
+                  ) : (
+                    <>
+                      <input
+                        aria-label={`${choice.name} Votes`}
+                        className={classnames(
+                          Classes.INPUT,
+                          errors.results?.[choice.id] && Classes.INTENT_DANGER
+                        )}
+                        name={`results[${choice.id}]`}
+                        readOnly={isSubmitting}
+                        ref={register({
+                          min: 0,
+                          required: true,
+                          valueAsNumber: true,
+                        })}
+                        type="number"
+                        // Visually hide this input instead of completely unmounting it since
+                        // rendering no input at all interferes with react-hook-form's state
+                        // management, specifically its isDirty tracking
+                        style={{ display: !isEditing ? 'none' : undefined }}
+                      />
+                      {!isEditing && (
+                        <span>{selectedSheet?.results[choice.id] || 0}</span>
+                      )}
+                    </>
+                  )}
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </BatchResultTallySheetTable>
+
+        <BatchResultTallySheetButtonRow>
+          {(() => {
+            if (selectedTabId === VOTE_TOTALS_TAB_ID) {
+              return <span>Select a sheet to make edits.</span>
```

Comment:
I like the way you handled this case! Small copy tweak:

```suggestion
              return <span>Select a sheet to edit tallies</span>
```